### PR TITLE
[Optimize][Clone] Take version count into consideration when choosing src replica for clone task

### LINKED ISSUE #6512: [Optimize][Clone] Optimize clone task replica selector
If there is  missing replica for a tablet, clone task will be executed to restore missing replica from a healthy replica. Src replica selector will randomly choose a healthy replica as src replica.

 It's better to choose the health replica with minmun version count as src replica so that it could avoid repetitive compaction task. In addition, replica with less version count is good for query performance.

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

Fix #6512 

If there is missing replica for a tablet, clone task will be executed to restore missing replica from a healthy replica. Src replica selector will randomly choose a healthy replica as src replica.

It's better to choose the health replica with minmun version count as src replica so that it could avoid repetitive compaction task. In addition, replica with less version count is good for query performance.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

### CODE REVIEW

**Review by morningman — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java [hunk 1]
```diff
@@ -499,10 +501,26 @@ public void chooseSrcReplica(Map<Long, PathSlot> backendsWorkingSlots) throws Sc
...
+                    if (curVerCount < minVersionCount) {
+                        minVersionCount = curVerCount;
+                        setSrc(srcReplica);
+                        findSrcReplica = true;
+                    }
+                }
             }
         }
+
+        if (findSrcReplica) {
```
Comment (caiconghui) [inline fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java:521]:
```suggestion
        if (!findSrcReplica) {
             throw new SchedException(Status.SCHEDULE_FAILED, "unable to find source slot");
        }

       
```

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java [hunk 2]
```diff
@@ -499,10 +501,26 @@ public void chooseSrcReplica(Map<Long, PathSlot> backendsWorkingSlots) throws Sc
             
             long srcPathHash = slot.takeSlot(srcReplica.getPathHash());
             if (srcPathHash != -1) {
-                setSrc(srcReplica);
-                return;
+                if (!findSrcReplica) {
```
Comment (caiconghui) [inline fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java:504]: remove the first if block

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java (hunk 1 above)
Comment (weizuo93) [reply fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java:521]: It's necessary to record first candidate replica in the `if` block.

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java (hunk 2 above)
Comment (weizuo93) [reply fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java:504]: It's necessary to record first candidate replica in the `if` block.

**Pull request merged**